### PR TITLE
Adding CloudSQL Functionality

### PR DESCRIPTION
**Pull request by GreenStorm-Code**
Added cloudsql functionality to list cloudsql instances, including all tests

### CODE REVIEW

File: tools/cli.py
Comment (Fryyyyy) [inline tools/cli.py:344]:
```suggestion
  AddParser('gcp', gcp_subparsers, 'listcloudsqlinstances', 'List CloudSQL instances for a project.')
```

File: tools/gcp_cli.py
Comment (Fryyyyy) [inline tools/gcp_cli.py:260]:
```suggestion
    logger.info('{0:s} {1:s} [{2:s}]'.format(
```

File: tests/providers/gcp/gcp_test.py
Comment (Fryyyyy) [inline tests/providers/gcp/gcp_test.py:1]: This file no longer exists since the tests were refactored in e0ccf13185d1a5002e82a8b23a2d4ba09307aaf2

File: libcloudforensics/providers/gcp/internal/cloudsql.py
Comment (Fryyyyy) [inline libcloudforensics/providers/gcp/internal/cloudsql.py:24]: SQL is an acronym so should probably be GoogleCloudSQL everywhere in names (as we have done in comments)

**Review by Fryyyyy — COMMENTED** [review]
At the moment tests aren't passing

File: libcloudforensics/providers/gcp/internal/cloudsql.py
Comment (giovannt0) [inline libcloudforensics/providers/gcp/internal/cloudsql.py:60]: Missing return type: `List[Dict[str, Any]]: List of instances.`

**Review by giovannt0 — APPROVED** [review]
Small nit, otherwise LGTM

**Comment by tomchop** [comment]
@GreenStorm-Code Just FYI, this is with you still. Feel free to make the requested changes so that we can merge the PR 👍

**Comment by Fryyyyy** [comment]
LGTM once the current comments are fixed and merge conflicts resolved

**Review by Fryyyyy — APPROVED** [review]
LGTM once the current comments are fixed and merge conflicts resolved

**Comment by tomchop** [comment]
@GreenStorm-Code could you review the merge conflicts here and address comments so we can merge this? Thanks! :)

**Comment by giovannt0** [comment]
@googlebot I consent.

**Pull request merged**
